From f79693658be2e3fdc9d1c2d1a112f92d5e22e303 Mon Sep 17 00:00:00 2001
From: Richard Levitte <levitte@openssl.org>
Date: Wed, 5 Jun 2024 21:43:01 +0200
Subject: [PATCH 007/125] Configure: make absolutedir() use rel2abs() on
 Windows too

perl's realpath() seems to be buggy on Windows, so we turn to rel2abs()
there as well.

Fixes #23593

Reviewed-by: Neil Horman <nhorman@openssl.org>
Reviewed-by: Tomas Mraz <tomas@openssl.org>
(Merged from https://github.com/openssl/openssl/pull/24569)

(cherry picked from commit 6e01d3114b77c82cf83a2bfe53f7ba97840fbe36)
---
 Configure | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/Configure b/Configure
index d1c3faa1a3..86190d064d 100755
--- a/Configure
+++ b/Configure
@@ -3495,6 +3495,13 @@ sub absolutedir {
         return rel2abs($dir);
     }
 
+    # realpath() on Windows seems to check if the directory actually exists,
+    # which isn't what is wanted here.  All we want to know is if a directory
+    # spec is absolute, not if it exists.
+    if ($^O eq "MSWin32") {
+        return rel2abs($dir);
+    }
+
     # We use realpath() on Unix, since no other will properly clean out
     # a directory spec.
     use Cwd qw/realpath/;
-- 
2.39.5 (Apple Git-154)

